src/Core/View.js

```diff
@@ -107,8 +107,21 @@ function _preprocessLayer(view, layer, provider) {
             layer.tileTextureCount = provider.tileTextureCount.bind(provider);
         }
 
-        if (provider.preprocessDataLayer) {
-            provider.preprocessDataLayer(layer);
+        if (!layer.preprocessedPromise) {
```

Comment:
`onFoo` is traditionnally more for event. Maybe `whenReady`? It will read "layer when ready then".